lib/Canonicalize/SimplifySwitch/SimplifySwitchPipe.cpp

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include "llvm/IR/LegacyPassManager.h"
+
+#include "revng/Lift/LoadBinaryPass.h"
+#include "revng/Model/Binary.h"
+#include "revng/Model/RawBinaryView.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Contract.h"
+#include "revng/Pipeline/Kind.h"
+#include "revng/Pipeline/LLVMContainer.h"
+#include "revng/Pipeline/RegisterPipe.h"
+#include "revng/Pipes/FileContainer.h"
+
+#include "revng-c/Pipes/Kinds.h"
+
+#include "SimplifySwitchPass.h"
+
+namespace revng::pipes {
+
+class SimplifySwitch {
+public:
+  static constexpr auto Name = "simplify-switch";
+
+  std::array<pipeline::ContractGroup, 1> getContract() const {
+    // // TODO: Is an empty contract valid?
+    return { pipeline::ContractGroup({}) };
```

Comment:
1. Don't use `TODO` for things that are broken, use `WIP` so we're sure this doesn't go in.
2. @drblallo this pipe reads the model, the binary and makes some changes to the IR. How should the contract look like?

Reply:
@dwarfyresearch [here's an example](https://github.com/revng/revng-c/blob/9624f2bbd14b6a3aead30f7c96fc6ed9a0b255ad/lib/RemoveLiftingArtifacts/MakeSegmentRefPipe.cpp#L26C1-L33C4) of what it should look like. Make sure the kind of the functions is the correct one.